lib/shopify_cli/context.rb

```diff
@@ -619,11 +620,13 @@ def which(cmd)
     #            : nil otherwise
     #
     def new_version
-      if (time_of_last_check + VERSION_CHECK_INTERVAL) < (now = Time.now.to_i)
-        update_time_of_last_check(now)
-        latest_version = retrieve_latest_gem_version
-        latest_version unless latest_version == ShopifyCLI::VERSION
+      if (time_of_last_check + VERSION_CHECK_INTERVAL) < (Time.now.to_i)
+        fork do
+          retrieve_latest_gem_version
+        end
       end
+      latest_version = ShopifyCLI::Config.get(VERSION_CHECK_SECTION, LATEST_VERSION_FIELD, default: ShopifyCLI::VERSION)
+      latest_version unless latest_version == ShopifyCLI::VERSION
```

Comment:
This will always return the `latest_version` even if `(time_of_last_check + VERSION_CHECK_INTERVAL) < (now = Time.now.to_i)` and it should have never been checked in the first place.